src/Opdex.Platform.Application/EntryHandlers/VaultGovernances/Proposals/GetVaultProposalsWithFilterQueryHandler.cs

```diff
@@ -0,0 +1,40 @@
+using MediatR;
+using Opdex.Platform.Application.Abstractions.EntryQueries.VaultGovernances.Proposals;
+using Opdex.Platform.Application.Abstractions.Models.VaultGovernances;
+using Opdex.Platform.Application.Abstractions.Queries.VaultGovernances;
+using Opdex.Platform.Application.Abstractions.Queries.VaultGovernances.Proposals;
+using Opdex.Platform.Application.Assemblers;
+using Opdex.Platform.Domain.Models.VaultGovernances;
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Opdex.Platform.Application.EntryHandlers.VaultGovernances.Proposals;
+
+public class GetVaultProposalsWithFilterQueryHandler : EntryFilterQueryHandler<GetVaultProposalsWithFilterQuery, VaultProposalsDto>
+{
+    private readonly IMediator _mediator;
+    private readonly IModelAssembler<VaultProposal, VaultProposalDto> _proposalAssembler;
+
+    public GetVaultProposalsWithFilterQueryHandler(IMediator mediator, IModelAssembler<VaultProposal, VaultProposalDto> proposalAssembler)
+    {
+        _mediator = mediator ?? throw new ArgumentNullException(nameof(mediator));
+        _proposalAssembler = proposalAssembler ?? throw new ArgumentNullException(nameof(proposalAssembler));
+    }
+
+    public override async Task<VaultProposalsDto> Handle(GetVaultProposalsWithFilterQuery request, CancellationToken cancellationToken)
+    {
+        var vault = await _mediator.Send(new RetrieveVaultGovernanceByAddressQuery(request.Vault, findOrThrow: true), cancellationToken);
+
+        var proposals = await _mediator.Send(new RetrieveVaultProposalsWithFilterQuery(vault.Id, request.Cursor), cancellationToken);
+
+        var proposalsResults = proposals.ToList();
+
+        var cursorDto = BuildCursorDto(proposalsResults, request.Cursor, pointerSelector: result => result.PublicId);
```

Comment:
We may come back to this but I'm ok with this for now. 
We might want to sort by (ExpirationDate, PublicId) so we always get the active proposals first that aren't expired.

PublicId will do fine most of the time, but with expiration times changing based on Proposal Status, it may not always be the case

Reply:
Originally I had the cursor on `(ExpirationDate, PublicId)` but didn't understand the reason why it should be done this way. This explanation clarifies it, so I'll amend it back to how it was